frontend/src/pages/modelRegistry/screens/ModelRegistry.tsx

```diff
@@ -2,21 +2,31 @@ import React from 'react';
 import ApplicationsPage from '~/pages/ApplicationsPage';
 import { ProjectObjectType } from '~/concepts/design/utils';
 import TitleWithIcon from '~/concepts/design/TitleWithIcon';
-import ModelRegistryEmpty from '~/pages/modelRegistry/ModelRegistryEmpty';
 import { ModelRegistryContext } from '~/concepts/modelRegistry/context/ModelRegistryContext';
 import useRegisteredModels from '~/concepts/modelRegistry/apiHooks/useRegisteredModels';
 import RegisteredModelListView from './RegisteredModelListView';
+import EmptyRegisteredModels from './EmptyRegisteredModels';
+import RegisteredModelTableToolbar from './RegisteredModelsTableToolbar';
 
 const ModelRegistry: React.FC = () => {
-  const { modelRegistries } = React.useContext(ModelRegistryContext);
+  const { preferredModelRegistry } = React.useContext(ModelRegistryContext);
   const [registeredModels, loaded, loadError] = useRegisteredModels();
 
   return (
     <ApplicationsPage
-      empty={modelRegistries.length === 0}
-      emptyStatePage={<ModelRegistryEmpty />}
+      empty={registeredModels.size === 0}
+      emptyStatePage={
+        <>
+          <RegisteredModelTableToolbar />
+          <EmptyRegisteredModels preferredModelRegistry={preferredModelRegistry?.metadata.name} />
+        </>
+      }
       title={
-        <TitleWithIcon title="Registered models" objectType={ProjectObjectType.registeredModels} />
+        <TitleWithIcon
+          data-testid="app-page-title"
```

Comment:
Make this id specific to registered models, like `registered-model-page-title`

Reply:
There were some failing tests due to which I added this. Removed it now.